File: java/src/com/wildlava/explore/ItemContainer.java
//
// Explore
//    - The Adventure Interpreter
//
// Copyright (C) 2006  Joe Peterson
//

package com.wildlava.explore;

import java.util.ArrayList;

class ItemContainer
{
   World world;

   ArrayList<String> items;
   int item_limit = -1;

   ItemContainer(World world)
   {
      this.world = world;

      items = new ArrayList<String>();
   }

   String expandItemName(String item)
   {
      String same_item = null;

      if (world.same_items.containsKey(item))
      {
         same_item = world.same_items.get(item);
      }

      for (int i=0; i<items.size(); ++i)
      {
         String test_item = items.get(i);

         if (test_item.equals(item))
         {
            return test_item;
         }

         if (same_item != null && test_item.equals(same_item))
         {
            return test_item;
         }

         String[] word_list = ExpUtil.parseToArray(test_item, " ");
         if (word_list.length > 1)
         {
            if (word_list[0].equals(item) ||
                word_list[word_list.length - 1].equals(item))
            {
               return test_item;
            }
         }
      }

      return item;
   }

   boolean hasNoItems()
   {
      return items.isEmpty();
   }

   boolean hasItem(String item)
   {
      return items.contains(item);
   }

   boolean isFull()
   {
      return ((item_limit != -1) && (items.size() >= item_limit));
   }

   boolean addItem(String item, boolean no_limit)
   {
      if (isFull() && !no_limit)
      {
         return false;
      }

      items.add(new String(item));

      return true;
   }

   void setItemLimit(int n)
   {
      item_limit = n;
   }

   boolean removeItem(String item)
   {
      if (items.contains(item))
      {
         items.remove(items.indexOf(item));
         return true;
      }

      return false;
   }
}

File: python/modular/ItemContainer.py
#
# Explore
#    - The Adventure Interpreter
#
# Copyright (C) 2006  Joe Peterson
#

class ItemContainer:
    def __init__(self):
        self.items = []
        self.item_limit = None
        
    def has_no_items(self):
        return len(self.items) == 0
    
    def has_item(self, item):
        return item in self.items

    def is_full(self):
        if self.item_limit == None or len(self.items) < self.item_limit:
            return False
        else:
            return True
    
    def expand_item_name(self, item):
        if item in self.items:
            return item
            
        for test_item in self.items:
            word_list = test_item.split()
            if len(word_list) > 1:
                if word_list[0] == item or word_list[-1] == item:
                    return test_item
            
        return item

    def add_item(self, item, mayExpand):
        if not self.is_full() or mayExpand:
            self.items.append(item)
            return True
        else:
            return False

    def remove_item(self, item):
        if item in self.items:
            self.items.remove(item)
            return True
        else:
            return False
